src/main/scala/io/iohk/ethereum/consensus/pow/EthashDAGManager.scala

```diff
@@ -0,0 +1,94 @@
+package io.iohk.ethereum.consensus.pow
+
+import akka.util.ByteString
+import io.iohk.ethereum.consensus.pow.EthashMiner.DagFilePrefix
+import io.iohk.ethereum.utils.{ByteUtils, Logger}
+import org.bouncycastle.util.encoders.Hex
+
+import java.io.{File, FileInputStream, FileOutputStream}
+import scala.util.{Failure, Success, Try}
+
+class EthashDAGManager(blockCreator: EthashBlockCreator) extends Logger {
+  var currentEpoch: Option[Long] = None
+  var currentEpochDagSize: Option[Long] = None
+  var currentEpochDag: Option[Array[Array[Int]]] = None
```

Comment:
are vars unavoidable here?

Reply:
This code is considered stable and working, it's complete out of the current scope doing any modification to it.
In the best case I can add val to replace the magic numbers you mentioned. The implementation of the ethash computation is based on this: https://eth.wiki/concepts/ethash/ethash